mm2src/coins/lightning/ln_utils.rs

```diff
@@ -215,30 +103,40 @@ pub struct LightningParams {
     pub node_name: [u8; 32],
     // Node's RGB color. This is used for showing the node in a network graph with the desired color.
     pub node_color: [u8; 3],
+    // Invoice Payer is initialized while starting the lightning node, and it requires the number of payment retries that
+    // it should do before considering a payment failed or partially failed. If not provided the number of retries will be 5
+    // as this is a good default value.
+    pub payment_retries: Option<usize>,
+    // Node's backup path for channels and other data that requires backup.
+    pub backup_path: Option<String>,
 }
 
-#[cfg(target_arch = "wasm32")]
-pub async fn start_lightning(
-    _ctx: &MmArc,
-    _platform_coin: UtxoStandardCoin,
-    _ticker: String,
-    _params: LightningParams,
-    _network: Network,
-) -> EnableLightningResult<LightningCoin> {
-    MmError::err(EnableLightningError::UnsupportedMode(
-        "'connect_to_lightning_node'".into(),
-        "native".into(),
-    ))
+pub fn ln_data_dir(ctx: &MmArc, ticker: &str) -> PathBuf { ctx.dbdir().join("LIGHTNING").join(ticker) }
+
+pub fn ln_data_backup_dir(ctx: &MmArc, path: Option<String>, ticker: &str) -> Option<PathBuf> {
+    path.map(|p| {
+        PathBuf::from(&p)
+            .join(&hex::encode(&**ctx.rmd160()))
+            .join("LIGHTNING")
+            .join(ticker)
+    })
 }
 
-#[cfg(not(target_arch = "wasm32"))]
 pub async fn start_lightning(
     ctx: &MmArc,
     platform_coin: UtxoStandardCoin,
-    ticker: String,
+    protocol_conf: LightningProtocolConf,
+    conf: LightningCoinConf,
     params: LightningParams,
-    network: Network,
 ) -> EnableLightningResult<LightningCoin> {
+    // Todo: add support for Hardware wallets for funding transactions and spending spendable outputs (channel closing transactions)
```

Comment:
This fn is over 300 lines long. Please refactor it.

Reply:
Sure :), always planned to do this but didn't get to it.